Rakefile

```diff
@@ -57,24 +57,18 @@ end
 task :build, [:build_fat] do |t, args|
   args.with_defaults(:build_fat => false)
   print_info "Building project (fat: #{args[:build_fat]})"
-  sh %Q(swift build -c release --arch x86_64 --arch arm64 --disable-sandbox --build-path #{BUILD_DIR})
+
+  # Use xcodebuild due to https://bugs.swift.org/browse/SR-15802
+  xcpretty %Q(xcodebuild -resolvePackageDependencies)
+  xcpretty %Q(xcodebuild -scheme sourcery -destination platform=macOS,arch=x86_64 -derivedDataPath #{BUILD_DIR}derivedData -archivePath #{BUILD_DIR}sourcery.xcarchive archive)
```

Comment:
So the one downside to using a static library distributed via XCFramework's is [SR-15802](https://bugs.swift.org/browse/SR-15802). `swift build`/`swift run` will work fine, but specifying `--arch x86_64 --arch arm64` will result in duplicate symbols. 

To work around this, I updated the `build` task to use `xcodebuild` instead. Two things though: 

1. Are you happy with this? I don't know if we need to be able to build a fat binary anywhere else? Will it cause problems elsewhere? Is it worth it? 
2. I'm not sure if its my Apple Silicon machine or not, but I found the command a little unreliable with Xcode 13.2.1 related to the `-destination` specifier... Worth trying your end, but I can provide more info if necessary. I have not checked with Xcode 13.3 yet.

Reply:
Ah ok, so CI also got caught up by the second issue too, so it's not just me...

> ```
> set -o pipefail && xcodebuild -resolvePackageDependencies | xcpretty -c
> set -o pipefail && xcodebuild -scheme sourcery -destination platform=macOS,arch=x86_64 -derivedDataPath build/derivedData -archivePath build/sourcery.xcarchive archive | xcpretty -c
> 2022-03-15 18:25:27.041 xcodebuild[1789:175189] [MT] IDEFileReferenceDebug: [Load] <IDESwiftPackageCore.IDESwiftPackageDependencyFileReference, 0x7fcb7d261120: name:Quick path:absolute:/Users/distiller/project/build/derivedData/SourcePackages/checkouts/Quick> Failed to load container at path: /Users/distiller/project/build/derivedData/SourcePackages/checkouts/Quick, Error: Error Domain=com.apple.dt.IDEContainerErrorDomain Code=6 "Cannot open "Quick" as a "Swift Package Proxy" because it is already open as a "Swift User Managed Package Folder"." UserInfo={NSLocalizedDescription=Cannot open "Quick" as a "Swift Package Proxy" because it is already open as a "Swift User Managed Package Folder".}
> 2022-03-15 18:25:27.050 xcodebuild[1789:175189] [MT] IDEFileReferenceDebug: [Load] <IDESwiftPackageCore.IDESwiftPackageDependencyFileReference, 0x7fcb7d24d190: name:Stencil path:absolute:/Users/distiller/project/build/derivedData/SourcePackages/checkouts/Stencil> Failed to load container at path: /Users/distiller/project/build/derivedData/SourcePackages/checkouts/Stencil, Error: Error Domain=com.apple.dt.IDEContainerErrorDomain Code=6 "Cannot open "Stencil" as a "Swift Package Proxy" because it is already open as a "Swift User Managed Package Folder"." UserInfo={NSLocalizedDescription=Cannot open "Stencil" as a "Swift Package Proxy" because it is already open as a "Swift User Managed Package Folder".}
> 2022-03-15 18:25:27.181 xcodebuild[1789:175189] [MT] IDEFileReferenceDebug: [Load] <IDESwiftPackageCore.IDESwiftPackageDependencyFileReference, 0x7fcb7d513c10: name:Spectre path:absolute:/Users/distiller/project/build/derivedData/SourcePackages/checkouts/Spectre> Failed to load container at path: /Users/distiller/project/build/derivedData/SourcePackages/checkouts/Spectre, Error: Error Domain=com.apple.dt.IDEContainerErrorDomain Code=6 "Cannot open "Spectre" as a "Swift Package Proxy" because it is already open as a "Swift User Managed Package Folder"." UserInfo={NSLocalizedDescription=Cannot open "Spectre" as a "Swift Package Proxy" because it is already open as a "Swift User Managed Package Folder".}
> 2022-03-15 18:25:27.204 xcodebuild[1789:175189] [MT] IDEFileReferenceDebug: [Load] <IDESwiftPackageCore.IDESwiftPackageDependencyFileReference, 0x7fcb76dc3170: name:SwiftSyntax path:absolute:/Users/distiller/project/build/derivedData/SourcePackages/checkouts/swift-syntax> Failed to load container at path: /Users/distiller/project/build/derivedData/SourcePackages/checkouts/swift-syntax, Error: Error Domain=com.apple.dt.IDEContainerErrorDomain Code=6 "Cannot open "swift-syntax" as a "Swift Package Proxy" because it is already open as a "Swift User Managed Package Folder"." UserInfo={NSLocalizedDescription=Cannot open "swift-syntax" as a "Swift Package Proxy" because it is already open as a "Swift User Managed Package Folder".}
> 2022-03-15 18:25:27.921 xcodebuild[1789:175189] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-19234/IDEFoundation/Execution/RunContextManager/IDERunContextManager.m:823
> Details:  Error deleting scheme: Cannot modify data because the process disallows saving.
> Object:   <IDERunContextManager: 0x7fcb74d9cc50>
> Method:   -deleteRunContexts:completionQueue:completionBlock:
> Thread:   <NSThread: 0x7fcb73016e90>{number = 1, name = main}
> Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
> 2022-03-15 18:25:27.922 xcodebuild[1789:175189] could not delete old scheme: Error Domain=DVTCustomDataStoringErrors Code=0 "Cannot modify data because the process disallows saving." UserInfo={NSLocalizedDescription=Cannot modify data because the process disallows saving.}
> 2022-03-15 18:25:27.922 xcodebuild[1789:175189] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-19234/IDEFoundation/Execution/RunContextManager/IDERunContextManager.m:823
> Details:  Error deleting scheme: Cannot modify data because the process disallows saving.
> Object:   <IDERunContextManager: 0x7fcb74d9cc50>
> Method:   -deleteRunContexts:completionQueue:completionBlock:
> Thread:   <NSThread: 0x7fcb73016e90>{number = 1, name = main}
> Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
> 2022-03-15 18:25:27.922 xcodebuild[1789:175189] could not delete old scheme: Error Domain=DVTCustomDataStoringErrors Code=0 "Cannot modify data because the process disallows saving." UserInfo={NSLocalizedDescription=Cannot modify data because the process disallows saving.}
> 2022-03-15 18:25:27.922 xcodebuild[1789:175189] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-19234/IDEFoundation/Execution/RunContextManager/IDERunContextManager.m:823
> Details:  Error deleting scheme: Cannot modify data because the process disallows saving.
> Object:   <IDERunContextManager: 0x7fcb74d9cc50>
> Method:   -deleteRunContexts:completionQueue:completionBlock:
> Thread:   <NSThread: 0x7fcb73016e90>{number = 1, name = main}
> Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
> 2022-03-15 18:25:27.922 xcodebuild[1789:175189] could not delete old scheme: Error Domain=DVTCustomDataStoringErrors Code=0 "Cannot modify data because the process disallows saving." UserInfo={NSLocalizedDescription=Cannot modify data because the process disallows saving.}
> 2022-03-15 18:25:27.922 xcodebuild[1789:175189] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-19234/IDEFoundation/Execution/RunContextManager/IDERunContextManager.m:823
> Details:  Error deleting scheme: Cannot modify data because the process disallows saving.
> Object:   <IDERunContextManager: 0x7fcb74d9cc50>
> Method:   -deleteRunContexts:completionQueue:completionBlock:
> Thread:   <NSThread: 0x7fcb73016e90>{number = 1, name = main}
> Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
> 2022-03-15 18:25:27.923 xcodebuild[1789:175189] could not delete old scheme: Error Domain=DVTCustomDataStoringErrors Code=0 "Cannot modify data because the process disallows saving." UserInfo={NSLocalizedDescription=Cannot modify data because the process disallows saving.}
> 2022-03-15 18:25:27.923 xcodebuild[1789:175189] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-19234/IDEFoundation/Execution/RunContextManager/IDERunContextManager.m:823
> Details:  Error deleting scheme: Cannot modify data because the process disallows saving.
> Object:   <IDERunContextManager: 0x7fcb74d9cc50>
> Method:   -deleteRunContexts:completionQueue:completionBlock:
> Thread:   <NSThread: 0x7fcb73016e90>{number = 1, name = main}
> Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
> 2022-03-15 18:25:27.923 xcodebuild[1789:175189] could not delete old scheme: Error Domain=DVTCustomDataStoringErrors Code=0 "Cannot modify data because the process disallows saving." UserInfo={NSLocalizedDescription=Cannot modify data because the process disallows saving.}
> 2022-03-15 18:25:27.923 xcodebuild[1789:175189] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-19234/IDEFoundation/Execution/RunContextManager/IDERunContextManager.m:823
> Details:  Error deleting scheme: Cannot modify data because the process disallows saving.
> Object:   <IDERunContextManager: 0x7fcb74d9cc50>
> Method:   -deleteRunContexts:completionQueue:completionBlock:
> Thread:   <NSThread: 0x7fcb73016e90>{number = 1, name = main}
> Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
> 2022-03-15 18:25:27.923 xcodebuild[1789:175189] could not delete old scheme: Error Domain=DVTCustomDataStoringErrors Code=0 "Cannot modify data because the process disallows saving." UserInfo={NSLocalizedDescription=Cannot modify data because the process disallows saving.}
> 2022-03-15 18:25:27.923 xcodebuild[1789:175189] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-19234/IDEFoundation/Execution/RunContextManager/IDERunContextManager.m:823
> Details:  Error deleting scheme: Cannot modify data because the process disallows saving.
> Object:   <IDERunContextManager: 0x7fcb74d9cc50>
> Method:   -deleteRunContexts:completionQueue:completionBlock:
> Thread:   <NSThread: 0x7fcb73016e90>{number = 1, name = main}
> Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
> 2022-03-15 18:25:27.923 xcodebuild[1789:175189] could not delete old scheme: Error Domain=DVTCustomDataStoringErrors Code=0 "Cannot modify data because the process disallows saving." UserInfo={NSLocalizedDescription=Cannot modify data because the process disallows saving.}
> xcodebuild: error: The workspace named "project" does not contain a scheme named "sourcery". The "-list" option can be used to find the names of the schemes in the workspace.
> rake aborted!
> ```

It's strange because `xcodebuild -list` does output `sourcery` as a valid option, and it works on a re-run. I think its something to do with the way that `xcodebuild` resolves SPM packages but I'm not entirely sure if its a bug, or me doing it wrong. I'll keep playing with it.

Reply:
Ahh, looks like I was missing the `generic/` specifier on the destination - https://mokacoding.com/blog/xcodebuild-destination-options/ saved me 🙂